src/updater.h

```diff
@@ -46,15 +47,15 @@ class MetadataUpdater {
     MetadataAgent::Metadata metadata;
   };
 
-  MetadataUpdater(MetadataAgent* store);
+  MetadataUpdater(MetadataAgent* store, MetadataReader* reader, std::string name);
   virtual ~MetadataUpdater();
 
   const MetadataAgentConfiguration& config() {
     return store_->config();
   }
 
   // Starts updating.
-  virtual void start() = 0;
+  virtual bool start() = 0;
```

Comment:
Why `bool`? If the configuration is invalid, a `start` is a no-op, isn't it?

Reply:
I needed a signal for the KubernetesUpdater to know that the parent updater had started successfully, before firing off the watch threads. The two options I thought of were return a bool or raise an exception in the PollingMetadataUpdater start. Thoughts?

Reply:
If it weren't for the desire to share a reader object, and the possible need to validate the watch and polling options together, I would have been considering splitting the watch and the poller into separate updaters.
I've already suggested using composition over inheritance here. But we could also have a `StartThreads` virtual function that actually starts the poller thread and is called from the (non-virtual) `start` after validation succeeds.
Thoughts?

Reply:
Discussed and addressed offline.